internal/gatewayapi/resource/load.go

```diff
@@ -129,11 +132,10 @@ func loadKubernetesYAMLToResources(input []byte, addMissingResources bool) (*Res
 				},
 				Spec: typedSpec.(gwapiv1.GatewayClassSpec),
 			}
-			// fill controller name by default controller name when gatewayclass controller name empty.
-			if addMissingResources && len(gatewayClass.Spec.ControllerName) == 0 {
-				gatewayClass.Spec.ControllerName = egv1a1.GatewayControllerName
```

Comment:
remove `fill gatewayclass controller name if is empty` since gatewayclass with empty controller name cannot pass local validator checks